proxy/http/HttpSM.cc

```diff
@@ -69,6 +70,7 @@
 #define USE_NEW_EMPTY_MIOBUFFER
 
 extern int cache_config_read_while_writer;
+extern std::unordered_map<std::string, std::string> TunnelMap; // stores the name of the servers to tunnel to
```

Comment:
It seems wrong to have this in so many places - is there not a flag that could be set earlier in the processing to avoid repeated lookups?

Reply:
well the lookup happens twice. First in the servername callback. second in the host setup code. the callback depends on the version of openssl and is not guaranteed to get triggered (in case CONNECT for e.g.,).

Reply:
but then again , it does look for the servername I am setting in sslvc. Let me think about it more